src/client/api/init.c

```diff
@@ -195,19 +199,29 @@ daos_init(void)
 	if (rc != 0)
 		D_GOTO(out_agent, rc);
 
+	D_ALLOC_PTR(info);
+	if (info == NULL)
+		D_GOTO(out_job, rc);
+	rc = dc_get_attach_info(NULL, true, info, &resp);
+	if (rc != 0) {
+		D_FREE(info);
+		D_GOTO(out_job, rc);
+	}
+	dc_set_global_attach_info(info, resp);
```

Comment:
[Suggestion] Would it be clearer if we move (the ownership of) the cached "info" and "resp" into `mgmt`? Then `info` and `resp` would no longer need to be global, and the cached "info" and "resp" would be freed when `daos_fini` calls `dc_clear_global_attach_info`.

Reply:
OK, ill look into on the next set of changes and rebase to this PR.

Reply:
ok i looked more into this, and dc_get_attach_info() expects the caller to own info and resp. and since we use them in init and fini, i had to make them global here.
I'll try and make this a little more clear.

Reply:
i reworked this a bit and ultimately did what you suggested with a new dc_mgmt function to populate those global variables that are owned in cli_mgmt.

Reply:
Thanks. Looks much clearer now.